base/sparse/sparsematrix.jl

```diff
@@ -9,10 +9,17 @@ type SparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
     colptr::Vector{Ti}      # Column i is in colptr[i]:(colptr[i+1]-1)
     rowval::Vector{Ti}      # Row values of nonzeros
     nzval::Vector{Tv}       # Nonzero values
-end
 
-SparseMatrixCSC{Tv,Ti}(m::Integer, n::Integer, colptr::Vector{Ti}, rowval::Vector{Ti}, nzval::Vector{Tv}) =
-    SparseMatrixCSC(int(m), int(n), colptr, rowval, nzval)
+    function SparseMatrixCSC(m::Integer, n::Integer, colptr::Vector{Ti}, rowval::Vector{Ti}, nzval::Vector{Tv})
+        ((m < 0) || (n < 0)) && error("invalid Array dimensions")
```

Comment:
I would write something like

`m < 0 && throw(ArgumentError("number of rows (m) must be ≥ 0, got $m"))`
`n < 0 && throw(ArgumentError("number of columns (n) must  be ≥ 0, got $n"))`